### PR TITLE
Set up functionality when 'Show all list details' is clicked

### LINKED ISSUE #44: Create functionality when 'Show all items details' is clicked
Need to create the full functionality of clicking 'Show all item details'. Upon clicking, it should expand all list items displaying their full details. The button should then switch to displaying 'Hide all item details' where users should then be able to click again to switch back to the original view of the list of items. 

## Acceptance Criteria
- [] clicking 'Show all item details' expands all list items displaying their full details
- [] clicking changes the text to 'Hide all item details'
- [] clicking 'Hide all item details' returns the list items back to their original view

### PR DESCRIPTION
**Pull request by neshacascia**
## Description
This feature builds out the full functionality of clicking 'Show all item details'. Upon clicking, it will expand all list items displaying their full details. The button will then switch to displaying 'Hide all item details' where users can click again to switch back to the original view of the list of items. 

## Related Issue

Closes #44  

## Acceptance Criteria
- [x] clicking 'Show all item details' expands all list items displaying their full details
- [x] clicking changes the text to 'Hide all item details'
- [x] clicking 'Hide all item details' returns the list items back to their original view 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before
Screen Shot 2023-03-03 at 10 17 27 PM

### After

https://user-images.githubusercontent.com/90223795/222873166-b0d2958c-1c08-47e9-b2b4-07a98e59cae7.mov

### CODE REVIEW

File: src/views/Layout.jsx [hunk 1]
```diff
@@ -11,19 +16,113 @@ import './Layout.css';
...
+										{data.length} items
+									</span>{' '}
+									in your shopping list
+								</p>
+								<div className="flex justify-center gap-2 mt-1">
+									<button
+										className="text-sm font-bold hover:text-main hover:cursor-pointer"
+										onClick={openAllItemDetails}
+									>
+										Show all item details
```
Comment (FuentesNina) [inline src/views/Layout.jsx]: @neshacascia  changing the text from show to hide based on state. 

```suggestion
										{allDetailsOpen ? 'Hide' : 'Show'} all item details
```

File: src/views/Layout.jsx [hunk 2]
```diff
@@ -11,19 +16,113 @@ import './Layout.css';
...
+									</button>
+									<div>|</div>
+									<Link
+										to="/about"
+										className="text-sm font-bold hover:text-main"
```
Comment (FuentesNina) [inline src/views/Layout.jsx:94]: @nesha centering the text vertically. 

```suggestion
										className="text-sm font-bold hover:text-main my-auto"
```

File: src/views/Layout.jsx [hunk 3]
```diff
@@ -11,19 +16,113 @@ import './Layout.css';
  * defined in `App.jsx`.
  */
 
-export function Layout() {
+export function Layout({ data, listToken, setAllDetailsOpen }) {
```
Comment (FuentesNina) [inline src/views/Layout.jsx]: @neshacascia  getting the prop for the current state

```suggestion
export function Layout({ data, listToken, setAllDetailsOpen, allDetailsOpen }) {
```

File: src/App.jsx [hunk 4]
```diff
@@ -50,7 +51,16 @@ export function App() {
...
 			<Router>
 				<Routes>
-					<Route path="/" element={<Layout />}>
+					<Route
+						path="/"
+						element={
+							<Layout
+								data={data}
+								listToken={listToken}
+								setAllDetailsOpen={setAllDetailsOpen}
```
Comment (FuentesNina) [inline src/App.jsx:61]: @neshacascia adding the prop for current state

```suggestion
								setAllDetailsOpen={setAllDetailsOpen}
								allDetailsOpen={allDetailsOpen}
```

**Review by FuentesNina — APPROVED** [review]
This works like a charm! Well done. I added a couple suggestions on changing the text from show to hide based on whether the item details are showing. And also, suggesting vertically centering the text for the about page link.

File: src/views/Layout.jsx (hunk 1 above)
Comment (neshacascia) [reply src/views/Layout.jsx]: Great addition, thanks Nina!

**Review by meganesu — APPROVED** [review]
Y'all got the swoopy corners working! This looks great, nice work! And the individual list item toggles work as expected when you use keyboard navigation :chefkiss: 🥳

**Pull request merged**